### ISSUE TITLE
Open source mobile-install v2

### ISSUE THREAD

**Issue by jin** [bazelbuild/bazel#6893]
mobile-install v2 is a Starlark rewrite of the native `mobile-install` command in Bazel. 

This is the tracking bug for the open sourcing of v2.

**Comment by steeve** [comment]
Will it possible to add iOS support via libimobiledevice? I made a working prototype (for device only)

**Comment by jin** [comment]
The current implementation is highly coupled to the transitive build graph from an `android_binary`, and uses aspects to traverse the graph to generate sharded application parts. 

I'm not familiar with libimobiledevice. However, the aspect approach should be fairly similar, although I would image you'll need a large rewrite, if not from scratch, to add iOS support.

**Comment by steeve** [comment]
Yeah I would figure the partial install wouldn't work. Although going from an `.ipa` to install is pretty straightforward via the `ideviceinstaller` CLI.

**Comment by jin** [comment]
`mobile-install` is currently stable in Bazel. I'm unassigning myself from this to focus my efforts on other issues.
